server/lib/manifest.py

```diff
@@ -0,0 +1,369 @@
+                provider_name = doc['meta']['provider']
+                if provider_name.startswith('HTTP'):
+                    provider_name = 'HTTP'  # TODO: handle HTTPS to make it unnecessary
+                provider = IMPORT_PROVIDERS.providerMap[provider_name]
+                top_identifier = provider.getDatasetUID(doc, self.user)
+                if top_identifier:
+                    dataset_top_identifiers.add(top_identifier)
+
+                ext_obj = {
+                    'dataset_identifier': top_identifier,
+                    'provider': provider_name,
+                    '_modelType': obj['_modelType'],
+                    'relpath': relpath
+                }
+
+                if obj['_modelType'] == 'folder':
+                    ext_obj['name'] = doc['name']
+
+                    if provider_name == 'HTTP':
+                        external_objects += _handle_http_folder(doc, self.user)
+                        continue
+
+                    if doc['meta'].get('identifier') == top_identifier:
+                        ext_obj['uri'] = top_identifier
+                    else:
+                        ext_obj['uri'] = provider.getURI(doc, self.user)
+                        #  Find path to root?
+
+                elif obj['_modelType'] == 'item':
+                    fileObj = self.itemModel.childFiles(doc)[0]
+                    ext_obj.update({
+                        'name': fileObj['name'],
+                        'uri': fileObj['linkUrl']
+                    })
+                external_objects.append(ext_obj)
+            except (ValidationException, KeyError):
+                msg = 'While creating a manifest for Tale "{}" '.format(str(self.tale['_id']))
+                msg += 'encountered a following error:\n'
+                logger.warning(msg)
+                raise  # We don't want broken manifests, do we?
+
+        return external_objects, dataset_top_identifiers
+
+    def add_dataset_records(self):
+        """
+        Adds dataset records to the manifest document
+        :return: None
+        """
+        for folder_id in self.datasets:
+            dataset_record = self.create_dataset_record(folder_id)
+            if dataset_record:
+                self.manifest['Datasets'].append(dataset_record)
+
+    def create_bundle(self, folder, filename):
+        """
+        Creates a bundle for an externally referenced file
+        :param folder: The name of the folder that the file is in
+        :param filename:  The name of the file
+        :return: A dictionary record of the bundle
+        """
+
+        # Add a trailing slash to the path if there isn't one (RO spec)
+        bundle = {}
+        if folder:
+            if not folder.endswith('/'):
+                folder += '/'
+            bundle['folder'] = folder
+        if filename:
+            bundle['filename'] = filename
+        return bundle
+
+    def add_system_files(self):
```

Comment:
It seems odd for the manifest endpoint to generate information about files that don't actually exist yet.  Including the ```schema:license``` makes sense, since we have the selected license, but the others won't be present until the ```tale_export``` branch is merged.   Should we remove this method for now?

Reply:
https://github.com/whole-tale/girder_wholetale/pull/251/commits/85b2c97983726c8fa975357715e34fb94677f8b3